braindecode/datasets/hub.py

```diff
@@ -0,0 +1,927 @@
+            if not zarr_path.exists():
+                raise FileNotFoundError(
+                    f"Zarr dataset not found at {zarr_path}. "
+                    "The dataset may be in an unsupported format."
+                )
+
+            dataset = cls._load_from_zarr_inline(zarr_path, preload)
+
+            log.info(f"Dataset loaded successfully from {repo_id}")
+            log.info(f"Recordings: {len(dataset.datasets)}")
+            log.info(
+                f"Total windows/samples: {format_info.get('total_samples', 'N/A')}"
+            )
+
+            return dataset
+
+        except huggingface_hub.utils.HfHubHTTPError as e:
+            if e.response.status_code == 404:
+                raise FileNotFoundError(
+                    f"Dataset '{repo_id}' not found on Hugging Face Hub. "
+                    "Please check the repository ID and ensure it exists."
+                )
+            else:
+                raise RuntimeError(f"Failed to download dataset: {e}")
+        except Exception as e:
+            raise RuntimeError(f"Failed to load dataset from Hub: {e}")
+
+    def _convert_to_zarr_inline(
+        self,
+        output_path: Path,
+        compression: str,
+        compression_level: int,
+    ) -> None:
+        """Convert dataset to Zarr format (inline implementation)."""
+
+        if zarr is False or huggingface_hub is False:
+            raise ImportError(
+                "huggingface hub functionality is not installed. Install with: "
+                "pip install braindecode[hub]"
+            )
+
+        if output_path.exists():
+            raise FileExistsError(
+                f"{output_path} already exists. Set overwrite=True to replace it."
+            )
+
+        # Create zarr store (zarr v3!)
+        store = zarr.storage.LocalStore(str(output_path))
+        root = zarr.group(store=store, overwrite=False)
+
+        # Validate uniformity across all datasets using shared validation
+        dataset_type, _, _ = hub_validation.validate_dataset_uniformity(self.datasets)
+
+        # Keep reference to first dataset for preprocessing kwargs
+        first_ds = self.datasets[0]
+
+        # Store global metadata
+        root.attrs["n_datasets"] = len(self.datasets)
+        root.attrs["dataset_type"] = dataset_type
+        root.attrs["braindecode_version"] = braindecode.__version__
+
+        # Save preprocessing kwargs (check first dataset, assuming uniform preprocessing)
+        # These are typically set by windowing functions on individual datasets
+        for kwarg_name in [
+            "raw_preproc_kwargs",
+            "window_kwargs",
+            "window_preproc_kwargs",
+        ]:
+            # Check first dataset for these attributes
+            if hasattr(first_ds, kwarg_name):
+                kwargs = getattr(first_ds, kwarg_name)
+                if kwargs:
+                    root.attrs[kwarg_name] = json.dumps(kwargs)
```

Comment:
@bruAristimunha the preprocessing steps are already saved here.
But, I just realise that it does not take into account the case when we apply some preprocessing on `RawDataset`, then convert it to `WindowDataset` (i.e. containing `mne.Epochs`). 
In this case, the steps applied on the raw data will be forgotten, only the steps applied on the windows will be there. It's something to modify in `create_windows_from_events` and other windowing functions